src/App.tsx

```diff
@@ -9,39 +9,33 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { postsSlice } from './features/posts';
+import * as postsActions from './features/posts';
+import { selectedPostSlice } from './features/selectedPost';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+  const {
+    items: posts,
+    loaded,
+    hasError,
+  } = useAppSelector(state => state.posts);
+  const selectedPost = useAppSelector(state => state.selectedPost);
+  const dispatch = useAppDispatch();
 
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
-
-  function loadUserPosts(userId: number) {
-    setLoaded(false);
-
-    getUserPosts(userId)
-      .then(setPosts)
-      .catch(() => setError(true))
-      // We disable the spinner in any case
-      .finally(() => setLoaded(true));
-  }
+  const author = useAppSelector(state => state.author);
 
   useEffect(() => {
     // we clear the post when an author is changed
     // not to confuse the user
-    setSelectedPost(null);
+    dispatch(selectedPostSlice.actions.setSelectedPost(null));
 
     if (author) {
-      loadUserPosts(author.id);
+      dispatch(postsActions.loadPosts(author.id));
     } else {
-      setPosts([]);
+      dispatch(postsSlice.actions.clearPosts());
     }
```

Comment:
In the `useEffect` hook, you're dispatching `postsSlice.actions.clearPosts()` when `author` is null. This could lead to unnecessary re-renders and affect performance. Consider adding a condition to check if there are any posts before dispatching this action.